crates/oxc_language_server/src/main.rs

```diff
@@ -33,13 +32,16 @@ mod linter;
 
 type ConcurrentHashMap<K, V> = papaya::HashMap<K, V, FxBuildHasher>;
 
+const OXC_CONFIG_FILE: &str = ".oxlintrc.json";
+
 struct Backend {
     client: Client,
     root_uri: OnceCell<Option<Url>>,
     server_linter: RwLock<ServerLinter>,
     diagnostics_report_map: ConcurrentHashMap<String, Vec<DiagnosticReport>>,
     options: Mutex<Options>,
     gitignore_glob: Mutex<Vec<Gitignore>>,
+    nested_configs: RwLock<FxHashMap<PathBuf, ConfigStore>>,
```

Comment:
I don't know if this is the correct type. `RwLock<FxHashMap<PathBuf, ConfigStore>>`. Maybe something else should be used, but this is the only one I could figure out how to make work.

Reply:
Seems ok to me, but I'm not an expert in the async data structures area 😄 Might be worth looking at https://docs.rs/papaya/latest/papaya/, as I know we have been using that recently?